text/0000-cfg-path-version.md

```diff
@@ -0,0 +1,956 @@
+`::core::ops::RangeInclusive` is defined to take advantage of this.
+
+So far we have only used `accessible(..)` to refer to paths in the standard 
+library. However, while it will be a less likely use case, you can use the flag
+to test if a path exists in some library in the ecosystem. This can for example
+be useful if you need to support lower minor versions of a library but also
+add support for features in a higher minor version.
+
+## `#[cfg(version(1.27.0))]`
+
+Until now, we have only improved our support for library features but never
+any language features. By checking if we are on a certain minimum version of
+Rust or any version above it, we can conditionally support such new features.
+For example:
+
+```rust
+#[cfg_attr(version(1.27), must_use)]
+fn double(x: i32) -> i32 {
+    2 * x
+}
+
+fn main() {
+    double(4);
+    // warning: unused return value of `double` which must be used
+    // ^--- This warning only happens if we are on Rust >= 1.27.
+}
+```
+
+Another example is opting into the system allocator on Rust 1.28 and beyond:
+
+```rust
+#[cfg(version(1.28))]
+// or: #[cfg(accessible(::std::alloc::System))]
+use std::alloc::System;
+
+#[cfg_attr(version(1.28), global_allocator)]
+static GLOBAL: System = System;
+
+fn main() {
+    let mut v = Vec::new();
+    // This will allocate memory using the system allocator.
+    // ^--- But only on Rust 1.28 and beyond!
+    v.push(1);
+}
+```
+
+Note that you won't be able to make use of `#[cfg(version(..))]` for these 
+particular features since they were introduced before this RFC's features
+get stabilized. This means that you can't for example add `version(1.28)`
+to your code and expect Rust 1.28 compilers to enable the code.
+However, there will be features in the future to use this mechanism on.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## `#[cfg(version(<semver>))]`
+
+To the `cfg` attribute, a `version` flag is added.
+This flag has the following grammar (where `\d` is any digit in `0` to `9`):
+
+```rust
+flag : "version" "(" semver ")" ;
+semver : digits ("." digits ("." digits)?)? ;
+digits : \d+ ;
+```
+
+[caret requirements]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements
+
+If and only if a Rust compiler considers itself to have a version which is
+greater or equal to the version in the `semver` string will the
+`#[cfg(version(<semver>)]` flag be considered active.
+Greater or equal is defined in terms of [caret requirements].
+
+## `#[cfg(accessible($path))]`
+
+To the `cfg` attribute, an `accessible` flag is added.
+
+### Syntactic form
+
+This flag requires that a `path` fragment be specified in it inside parenthesis
+but not inside a string literal. The `$path` must start with leading `::`
+and may not refer to any parts of the own crate (e.g. with `::crate::foo`,
+`::self::foo`, or `::super::foo` if such paths are legal).
```

Comment:
Does this imply that `#[cfg(accessible(path)]` can be used with dependencies other than `core` or `std`? For example, can I write `#[cfg(accessible(::lazy_static::lazy_static)]`?

Reply:
Yes, you can do that.